src/pages/student/categories/CategoryDetail/index.js

```diff
@@ -7,110 +8,52 @@ import SubcategoryCard from './components/SubcategoryCard';
 
 function Subcategories() {
   const [categories, setCategories] = useState(null);
-
-    
-  const Subcategories = [
-        
-    {
-            
-      id: 1,
-      title: 'Basic',
-      firstRow: 'Beginner',
-      secRow: '10 Quizzes',
-      thirRow: '6 Quizzes',
-    },
-    {
-      id: 2,
-      title: 'Arrays',
-      firstRow: 'Intermediate',
-      secRow: '10 Quizzes',
-      thirRow: '1 Quizzes',
-    },
-    {
-      id: 3,
-      title: 'DOM',
-      firstRow: 'Hard',
-      secRow: '10 Quizzes',
-      thirRow: '8 Quizzes',
-    },
-    {
-      id: 4,
-      title: 'Object-Oriented Programming',
-      firstRow: 'Intermediate',
-      secRow: '10 Quizzes',
-      thirRow: '5 Quizzes',
-    },
-    {
-      id: 5,
-      title: 'ECMAScript – ES6',
-      firstRow: 'Hard',
-      secRow: '10 Quizzes',
-      thirRow: '3 Quizzes',
-    },
-    {
-      id: 6,
-      title: 'Classes and Objects',
-      firstRow: 'Beginner',
-      secRow: '10 Quizzes',
-      thirRow: '5 Quizzes',
-    },
-    {
-      id: 7,
-      title: 'HTML',
-      firstRow: 'Intermediate',
-      secRow: '10 Quizzes',
-      thirRow: '3 Quizzes',
-    },
-    {
-      id: 8,
-      title: 'DOM',
-      firstRow: 'Beginner',
-      secRow: '10 Quizzes',
-      thirRow: '1 Quizzes',
-    },
-       
-  ];
+  const [category, setCategory] = useState(null);
+  const categoryId = useParams().id;
 
   useEffect(() => {
-    CategoryApi.getAll()
-      .then(({ data }) => {
+    CategoryApi.show({ categoryId }).then(({ data }) => {
+      setCategory(data.data);
+      CategoryApi.getAll({ category_id: categoryId }).then(({ data }) => {
         setCategories(data.data);
       });
+    });
   }, []);
 
   const renderCatList = () => {
-    return Subcategories.map((subcategory, idx) => {
-      return (
-        <SubcategoryCard key={idx} title={subcategory.title} firstRow={subcategory.firstRow} secRow={subcategory.secRow} thirRow={subcategory.thirRow}/>
-      );
-    });     
-        
+    return categories.map((subcategory, idx) => {
+      return <SubcategoryCard key={idx} category={subcategory} />;
+    });
   };
 
   return (
     <div>
       <div className={style.subTile}>
         <div>
-          <p className={style.title}>Web Development</p>
+          <p className={style.title}>{category?.name}</p>
         </div>
         <div id={style.quizlink}>
-          <a style={{color:'black', fontSize:'24px'}} href='categories/:id/quizzes' >Check Available Quizzes &gt;&gt;</a>
+          {category?.quizzes_count && (
```

Comment:
use this to prevent the zero value found in the rightmost corner of the page:
{category?.quizzes_count ? (
            <a
              style={{ color: 'black', fontSize: '24px' }}
              href={`/categories/${category?.id}/quizzes`}
            >
              Check Available Quizzes &gt;&gt;
            </a>
          ) : ''}

![Screenshot from 2021-11-19 15-58-18](https://user-images.githubusercontent.com/88467532/142586582-9cd27b61-3edb-4c2f-8cac-280867bc22ce.png)